utils/catalog-parser/src/main/java/de/sovity/edc/utils/catalog/mapper/DspDataOfferBuilder.java

```diff
@@ -65,7 +74,50 @@ private DspDataOffer buildDataOffer(JsonObject dataset) {
     }
 
     @NotNull
-    private DspContractOffer buildContractOffer(JsonObject json) {
-        return new DspContractOffer(JsonLdUtils.id(json), json);
+    DspContractOffer buildContractOffer(JsonObject json) {
+        /*
+         * /!\ Workaround
+         * TODO: can't reference a private repo in a public repo
+         * https://github.com/sovity/edc-broker-server-extension/issues/278
+         * https://github.com/sovity/edc-broker-server-extension/issues/409
+         *
+         * The Eclipse EDC uses a new random ID for each contract offer that it returns.
+         * This can't be used as an id.
+         * As a workaround, we must introduce our own ID.
+         * For a first iteration, we will assume that the content of the policy remains the same (same content, same order)
+         * and hash it to use it as a key.
+         */
+
+        String idFieldName = "@id";
+        val id = json.get(idFieldName);
+        val idAsString = (JsonString) id;
+        val parts = idAsString.getString().split(":");
+        if (parts.length != 3) {
+            throw new RuntimeException("Can't use " + idAsString + ": wrong format, must be made of 3 parts.");
+        }
+
+        val sw = new StringWriter();
+        try (val writer = Json.createWriter(sw)) {
```

Comment:
We have JsonUtils with functions like "toJson" and "fromJson" or sth like that to handle serialization and deserialization